### PR TITLE
Patch, instead of Update, node annotations

### PR DESCRIPTION
**Pull request by aanm**
~:warning: :warning: PR being merged against #7887 :warning: :warning:~

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7906)

### CODE REVIEW

**Review by ianvernon — APPROVED** [review]

File: pkg/k8s/annotate.go [hunk 1]
```diff
@@ -42,43 +43,44 @@ type K8sCiliumClient struct {
...
-	if node == nil {
-		return nil, ErrNilNode
+	raw, err := json.Marshal(annotations)
+	if err != nil {
+		return err
 	}
+	patch := []byte(fmt.Sprintf(`{"metadata":{"annotations":%s}}`, raw))
 
-	return node, nil
+	_, err = c.CoreV1().Nodes().Patch(nodeName, types.StrategicMergePatchType, patch)
```
Comment (tgraf) [inline pkg/k8s/annotate.go:81]: Why don't we need to check
```
type ServerCapabilities struct {
        // Patch is the ability to use PATCH to modify a resource
        Patch bool
```
?

**Review by tgraf — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: pkg/k8s/annotate.go (hunk 1 above)
Comment (aanm) [reply pkg/k8s/annotate.go:81]: Patch issues were only problematic for custom resources not k8s data structures. I test it with k8s 1.10 and worked fine.

**Review by tgraf — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Pull request merged**